File: Projection/demo/src/main/java/com/kuanquan/demo/DIDLLite.java
package com.kuanquan.demo;

import org.simpleframework.xml.Element;
import org.simpleframework.xml.Root;

/**
 * xml 解析框架生成的 bean
 */
@Root(name = "DIDL-Lite", strict = false)
public class DIDLLite {

    @Element(name = "item", required = false)
    public MediaItem item;

    @Root(name = "item", strict = false)
    public static class MediaItem {

        @Element(name = "class", required = false)
        public String objectClass;

        @Element(name = "title", required = false)
        public String title;

        @Element(name = "creator", required = false)
        public String creator;

        @Element(name = "album", required = false)
        public String album;

        @Element(name = "artist", required = false)
        public String artist;

        @Element(name = "albumArtURI", required = false)
        public String albumArtURI;
    }
}

File: Projection/test/src/main/java/com/kuanquan/test/projection/DIDLLite.kt
package com.kuanquan.test.projection

import org.simpleframework.xml.Element
import org.simpleframework.xml.Root
import java.io.Serializable

/**
 * xml 解析框架生成的 bean
 */
@Root(name = "DIDL-Lite", strict = false)
class DIDLLite: Serializable {

    @Element(name = "item", required = false)
    var item: MediaItem? = null

    @Root(name = "item", strict = false)
    class MediaItem: Serializable {
        @Element(name = "class", required = false)
        var objectClass: String? = null

        @Element(name = "title", required = false)
        var title: String? = null

        @Element(name = "creator", required = false)
        var creator: String? = null

        @Element(name = "album", required = false)
        var album: String? = null

        @Element(name = "artist", required = false)
        var artist: String? = null

        @Element(name = "albumArtURI", required = false)
        var albumArtURI: String? = null
    }
}